packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitHubAPI } from 'probot/lib/github';
+
+interface CheckParam {}
+
+interface CheckRun {
+  name: string;
+  conclusion: string;
+}
+
+interface CheckStatus {
+  context: string;
+  state: string;
+}
+interface Reviews {
+  user: {
+    login: string;
+  };
+  state: string;
+}
+
+interface ReviewsRequested {
+  users: [];
+  teams: [];
+}
+
+interface PR {
+  title: string;
+  body: string;
+}
+
+interface RequiredChecksByLanguage {
+  [key: string]: {
+    requiredStatusChecks: string[];
+  };
+}
+
+interface Language {
+  language: string;
+  repo: string;
+}
+
+// contains the installation id necessary to authenticate as an installation
+mergeOnGreen.getLatestCommit = async function getLatestCommit(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<string> {
+  try {
+    // TODO: consider switching this to an async iterator, which would work
+    // for more than 100.
+    const data = await github.pulls.listCommits({
+      owner,
+      repo,
+      pull_number: pr,
+      per_page: 100,
+      page: 1,
+    });
+    return data.data[data.data.length - 1].sha;
+  } catch (err) {
+    return '';
+  }
+};
+
+mergeOnGreen.getPR = async function getPR(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<PR> {
+  try {
+    const data = await github.pulls.get({
+      owner,
+      repo,
+      pull_number: pr,
+    });
+    return data.data;
+  } catch (err) {
+    return { title: '', body: '' };
+  }
+};
+
+mergeOnGreen.getMOGLabel = async function getMOGLabel(
+  owner: string,
+  repo: string,
+  pr: number,
+  labelName: string,
+  github: GitHubAPI
+): Promise<boolean> {
+  let isMOG = false;
+  try {
+    const labels = await github.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: pr,
+    });
+    const labelArray = labels.data;
+    if (labelArray) {
+      const mog = labelArray.find(element => element.name === labelName);
+      if (mog) {
+        isMOG = true;
+      }
+    }
+    return isMOG;
+  } catch (err) {
+    return isMOG;
+  }
+};
+
+mergeOnGreen.requiredChecksByLanguage = async function requiredChecksByLanguage(
+  github: GitHubAPI
+): Promise<RequiredChecksByLanguage | null> {
+  try {
+    const configFile = (
+      await github.repos.getContents({
+        owner: 'googleapis',
+        repo: 'sloth',
+        path: 'required-checks.json',
+      })
+    ).data as { content?: string };
+    return JSON.parse(
+      Buffer.from(configFile.content as string, 'base64').toString('utf8')
+    ) as RequiredChecksByLanguage;
+  } catch (err) {
+    return null;
+  }
+};
+
+mergeOnGreen.getRepoMap = async function getRepoMap(
```

Comment:
Looking just at the function name, I can't tell what this function is supposed to do. Can we make the name more descriptive or add documentation explaining what it's trying to do?

Reply:
Good call! Added a description and changed the name. Basically, I'm mapping the repo name to its language, since the sloth.json file only calls out required checks by language, not by repo